### PR TITLE
RESTWS-956 - Erroneous response when totalCount is included in query …

### PR DESCRIPTION
**Pull request by alaboso**
## Description of what I changed

I added `totalCount` as an optional parameter in EncounterSearchHandler2_0 to fix the error in [link](https://openmrs.atlassian.net/browse/RESTWS-956)

## Issue I worked on

https://openmrs.atlassian.net/browse/RESTWS-956

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java [hunk 1]
```diff
@@ -48,8 +48,8 @@ public class EncounterSearchHandler2_0 implements SearchHandler {
 			Collections.singletonList("2.0.* - 9.*"),
 			Collections.singletonList(new SearchQuery.Builder(
 					"Allows you to find Encounter by patient and encounterType (and optionally by from and to date range)")
-					.withRequiredParameters("patient").withOptionalParameters("visit", "encounterType", DATE_FROM, DATE_TO, "order")
-					.build()));
```
Comment (dkayiwa) [inline omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java:52]: Can we add a test for this?

Comment (alaboso) [reply omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java:52]: Sure, on it!

Comment (alaboso) [reply omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java:52]: I have added `EncounterSearchHandler2_0Test`

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0Test.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return "encounter";
+    }
+
+    /**
+     * @verifies returns encounters and totalCount filtered by patient uuid only
+     * @see @see EncounterSearchHandler2_0#search(RequestContext)
+     * @throws Exception
+     */
+    @Test
+    public void search_shouldReturnEncountersWithTotalCountFilteredByPatient() throws Exception {
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0Test.java:36]: Did you notice that this test passes even without your fix?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0Test.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * @verifies returns encounters and totalCount filtered by patient uuid and limit
+     * i.e. (limit 1, results size should not be the same as totalCount)
+     *
+     * @see @see EncounterSearchHandler2_0#search(RequestContext)
+     * @throws Exception
+     */
+    @Test
+    public void search_shouldReturnEncountersWithTotalCountFilteredByPatientLimitedToOne() throws Exception {
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0Test.java:79]: Did you notice that this test passes even without your fix?

**Comment by alaboso** [comment]
@dkayiwa  - I take note of your comments. The real test is this -> `search_shouldReturnEncountersWithTotalCountFilteredByPatientAndEncounterType` which checks that given and `encounterType` and `totalCount`, the results should be filtered. 

I could remove `search_shouldReturnEncountersWithTotalCountFilteredByPatient` though it clearly distinguishes the filtering...

In my opinion, `search_shouldReturnEncountersWithTotalCountFilteredByPatientLimitedToOne` should remain to clearly distinguish that the size of results returned is not necessarily the same as `totalCount` OR, add limit to `search_shouldReturnEncountersWithTotalCountFilteredByPatientAndEncounterType` after the current existing Assertions

**Pull request merged**